@xen-orchestra/lite/src/components/Tree.tsx

```diff
@@ -116,23 +129,66 @@ const Tree = withState<State, Props, Effects, Computed, ParentState, ParentEffec
   {
     initialState: ({ defaultSelectedNodes }) => ({
       selectedNodes: defaultSelectedNodes === undefined ? [] : defaultSelectedNodes,
+      _expandedNodes: undefined,
     }),
     effects: {
+      setExpandedNodeIds: function (event, nodeIds) {
+        this.state._expandedNodes = nodeIds
+      },
       setSelectedNodeIds: function (event, nodeIds) {
         this.state.selectedNodes = nodeIds
       },
     },
+    computed: {
+      expandedNodes: ({ _expandedNodes }, { collection, defaultSelectedNodes }) => {
```

Comment:
this operation is expensive, and when I add  a little debug I see that it's triggered a lot of time . Is it possible to check it  so that it's only used when needed?  

If it's not possible, is it possible to optimize it, for example with a map children -> node . That way instead of looking into each node of the tree you take the expanded children and can build the node chain faster